dask_ml/preprocessing/imputation.py

```diff
@@ -0,0 +1,90 @@
+from collections import OrderedDict
+
+from dask import persist
+from dask_ml.utils import slice_columns
+from sklearn.preprocessing import imputation as skimputation
+import dask.dataframe as dd
+import dask.array as da
+import numpy as np
+
+
+def _mask_values(X, missing_values):
+    """Compute the masked version of X."""
+    if missing_values == "NaN" or np.isnan(missing_values):
+        return X if isinstance(X, dd._Frame) else da.ma.masked_invalid(X)
+    else:
+        return (X.mask(X == missing_values) if isinstance(X, dd._Frame)
+                else da.ma.masked_equal(X, missing_values))
+
+
+def _safe_eq(X, v):
+    if isinstance(X, dd._Frame):
+        return X.isnull() if v == "NaN" or np.isnan(v) else X == v
+    else:
+        return da.isnull(X) if v == "NaN" or np.isnan(v) else X == v
+
+
+class Imputer(skimputation.Imputer):
+    def __init__(self, missing_values="NaN", strategy="mean",
+                 axis=0, verbose=0, copy=True, columns=None):
+        super().__init__(missing_values="NaN", strategy="mean",
+                         axis=0, verbose=0, copy=True)
+        self.columns = columns
+        self.missing_values = missing_values
```

Comment:
Does calling `super()` set all these for you?

Reply:
> Does calling super() set all these for you?

I'll check.

I was about to finish up the implementation of median when I noticed that if your series has `np.nan` values and you call `dd.Series.quantile` the output is also `np.nan` . Looks like dask doesn't drop them automatically (pandas does) @mrocklin .

I'm gonna insert a call to `dropna` before I compute the quantile, that should solve it for dataframes at least.

But yeah, I'll finish up and we can finish the other methods some other time. It would be nice to let people include any estimator as the `strategy` argument (as long as it is dask-compatible). I want to start working on the ALS estimator :)